lambdas/shared/t4_lambda_shared/preview.py

```diff
@@ -0,0 +1,55 @@
+"""
+Shared helper functions for generating previews for the preview lambda and the ES indexer.
+"""
+import zlib
+
+# MAX_BYTES is bytes scanned, so functions as an upper bound on bytes returned
+# in practice we will hit MAX_LINES first
+# we need a largish number for things like VCF where we will discard many bytes
+# Only applied to _from_stream() types. _to_memory types are size limited either
+# by pandas or by exclude_output='true'
+MAX_BYTES = 1024*1024
+MAX_LINES = 512 # must be positive int
+
+
+def get_preview_lines(chunk_iterator, compression, max_lines, max_bytes):
+    """
+    Read a (possibly compressed) text file, and return up to `max_lines` lines and `max_bytes` bytes.
+    """
+    buffer = []
+    size = 0
+    line_count = 0
+
+    if compression:
+        assert compression == 'gz', 'only gzip compression is supported'
```

Comment:
we don't need this because of https://github.com/quiltdata/quilt/pull/1079/files#diff-4ea57d41f3e2cb15d690feb1230f3d8dL63

Reply:
But, it's now used by the indexer, too - that's the whole point of this change.